lib/utilities/firebase_service.dart

```diff
@@ -64,13 +75,20 @@ class FirebaseService {
   }
 
   /// Creates game event in database
-  Future<String?> createEvent(SportEvent event) async {
+  Future<String?> createEvent(SportEvent event, String userId) async {
     try {
       var eventMap = event.toMap();
       final registrationDoc = _firestore.collection('events-collection').doc();
       await registrationDoc.set(eventMap);
 
-      return null;
+      final hostlogDoc = _firestore.collection('host').doc();
+      await hostlogDoc.set({
+        'userId' : userId,
+        'eventId' : registrationDoc.id,
+    });
+
+          print('${userId} created the event ${registrationDoc.id}');
+      
```

Comment:
What is the host collection used for? You should simply use the registration collection, like we already agreed. If you want to get the events that the current user is hosting, then you can query the event collection where hostUserId equals the ID of the logged in user. 
<img width="1317" alt="Screenshot 2024-12-02 at 5 34 54 PM" src="https://github.com/user-attachments/assets/24aefee4-f241-42f4-b21c-a1c7bbf68d7c">